generator/src/main/java/com/scottlogic/deg/generator/inputs/AtomicConstraintReaderLookup.java

```diff
@@ -30,14 +30,14 @@
                 (dto, fields, rules) ->
                     new FormatConstraint(
                         fields.getByName(dto.field),
-                        (String) throwIfValueInvalid(dto.value),
+                        throwIfValueInvalid(dto, String.class),
                         rules));
 
         add(AtomicConstraintType.ISEQUALTOCONSTANT.toString(),
                 (dto, fields, rules) ->
                     new IsInSetConstraint(
                         fields.getByName(dto.field),
-                        mapValues(Collections.singleton(throwIfValueInvalid(dto.value))),
+                        mapValues(Collections.singleton(throwIfValueInvalid(dto, Object.class))),
```

Comment:
Everything extends from `Object` therefore this method asserts only that `dto.value` has a value

Reply:
This GitHub comment helps clarify a lot - it's the sort of thing I'd love to see in the code :)

Reply:
Done